File: TwoPointer/26.remove-duplicates-from-sorted-array.go
package leetcode

// @leet start
func removeDuplicates(nums []int) int {
	l := 1
	for i, v := range nums {
		if i == 0 {
			continue
		}
		if v != nums[i-1] {
			nums[l] = v
			l++
		}
	}
	return l
}

// @leet end

func removeDuplicates(nums []int) int {
	l, r := 1, 1
	for r < len(nums) {
		if nums[r] == nums[r-1] {
			r++
			continue
		}
		nums[l] = nums[r]
		l++
		r++
	}
	return l
}

func removeDuplicates(nums []int) int {
	l, r := 1, 1
	for r < len(nums) {
		for r < len(nums) && nums[r] == nums[r-1] {
			r++
		} // need to keep check
		// when move slide window. just skip use one loop
		// avoid nested loop
		nums[l] = nums[r]
		l++
		r++
	}
	return l
}

File: TwoPointer/26.remove-duplicates-from-sorted-array.py
from typing import List


# @leet start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 1
        for r in range(1, len(nums)):
            if nums[r] == nums[r - 1]:
                continue
            nums[l] = nums[r]
            l += 1
        return l


# @leet end


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 1
        for r in range(1, len(nums)):
            if nums[r] != nums[r - 1]:
                nums[l] = nums[r]
                l += 1
        return l


def removeDuplicates(self, nums: List[int]) -> int:
    l = 0
    for v in nums:
        if nums[l] != v:
            nums[l] = v
            l += 1
    return l

